### ISSUE TITLE
Problem with bounding boxes visualization

### ISSUE THREAD

**Issue by federicocunico** [zju3dv/pvnet#56]
Hello,
I've tried to use the **pre-trained** cat model to infer the pose on all the LINEMOD's cat images using LINEMOD and LINEMOD_ORIG.
I followed the skeleton provided in tools/demo.py modifying demo() to loop for all the images and read_data() in order to get all the infos of the current index
I have encountered two problems:
1) in cat_points_3d.txt is contained the FPS 8 (furthest.txt) "plus the center", but I don't know how to actually **calculate** that "center" having a brand **new** 3d model/ply.
Can you tell me how to do it? Or how you calculated that 9th point in the first place? Because I see that there is no *_points_3d.txt for other models, just the furthest with only 8 points.

2) The first frame (the same as for demo) is ok in both gt bounding box and the infered one.
But starting from the second image, only the infered bounding box is placed in the right spot, the **ground truth it's not being placed on the cat at all**.
It seems very strange to me. I double checked everything, starting from which bounding box I was displaying up to each content of each file.
Do you have any clue?
I'll leave here a snapshot of how I got the poses that is a similar way as train script does.
```
def read_data(index):
      [...]
      # Get the rots and translations
      rots = get_files_from(os.path.join(dir_linemod_orig, obj_name, 'data'), contains='rot')
      tras = get_files_from(os.path.join(dir_linemod_orig, obj_name, 'data'), contains='tra')
      
      # Get pose from .rot and .tra
      pose = read_pose(rots[index], tras[index])

      # Transform pose from orig to blender
      pose_transformer = PoseTransformer(class_type=obj_name)
      pose = pose_transformer.orig_pose_to_blender_pose(pose).astype(np.float32)
      
      [...]
      data = (rgb, mask, vertex, vertex_weight, pose, points_2d)
      return return data, points_3d, bb8_3d
```

For demo purpose you provided the pose as .npy but in the LINEMOD_ORIG is given as .rot and .tra.
So I obtained the *.rot and *.tra files and I used the read_pose() function in lib.utils.data_utils. Then I used your orig_pose_to_blender_pose() in order to get at least the first image working, otherwise will do the first wrong too.
I may missing why you used that (doesn't the net learn about the numbers even if the axis are not inverted?).
Anyway after those steps, still the first image is the only one it's working properly.
Note that I used .rot and .tra even for the first frame. I am not using the .npy beacuse other frames does not have an npy file. Anyway the first frame all works fine with npy and also tra/rot files.

Thank you very much for the help!

First frame (same as demo)
![00001](https://user-images.githubusercontent.com/29544038/62623758-e6474300-b921-11e9-842a-53f9cf54e57f.png)

Other two random frames
![00002](https://user-images.githubusercontent.com/29544038/62623757-e6474300-b921-11e9-8869-042561fac5ee.png)
![00003](https://user-images.githubusercontent.com/29544038/62623756-e6474300-b921-11e9-967d-b63a11b5b3f6.png)

**Comment by pengsida** [comment]
The 3d model coordinate frame in the LINEMOD is different from that of LINEMOD_ORIG.
So we use `orig_pose_to_blender_pose` to transform the pose in LINEMOD_ORIG.

**Comment by federicocunico** [comment]
Oh ok! Thank you! So it is correct to use `orig_pose_to_blender_pose` using the LINEMOD_ORIG.
What about the 9th value, how to calculate the "center"? It's not the centroid otherwise you would call as it, I suppose, furthermore I tried to calculate the centroid of cat model and values does not match.

Thanks for help!

**Comment by pengsida** [comment]
The center is the mean value of 3D bounding box.
How did you calculate the centroid?

**Comment by federicocunico** [comment]
Ahh I used the 3d model of the cat, calculating the average value for each axis in order to get the centroid, I wasn't using the bounding box.
Thanks a lot

**Comment by pengsida** [comment]
Do you have correct visualization in other images now?

**Comment by federicocunico** [comment]
Yes, just checked, I'm closing the issue.
Thank you
